warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java

```diff
@@ -58,25 +62,96 @@ public BufferedFileBackedSortedSet(BufferedFileBackedSortedSet<E> other) {
         this.size = other.size;
     }
     
-    public BufferedFileBackedSortedSet(SortedSetFileHandlerFactory handlerFactory) {
-        this(null, DEFAULT_BUFFER_PERSIST_THRESHOLD, DEFAULT_MAX_OPEN_FILES, handlerFactory);
+    public BufferedFileBackedSortedSet(List<SortedSetFileHandlerFactory> handlerFactories) {
+        this(null, DEFAULT_BUFFER_PERSIST_THRESHOLD, DEFAULT_MAX_OPEN_FILES, DEFAULT_NUM_RETRIES, handlerFactories);
     }
     
-    public BufferedFileBackedSortedSet(Comparator<? super E> comparator, SortedSetFileHandlerFactory handlerFactory) {
-        this(comparator, DEFAULT_BUFFER_PERSIST_THRESHOLD, DEFAULT_MAX_OPEN_FILES, handlerFactory);
+    public BufferedFileBackedSortedSet(Comparator<? super E> comparator, List<SortedSetFileHandlerFactory> handlerFactories) {
+        this(comparator, DEFAULT_BUFFER_PERSIST_THRESHOLD, DEFAULT_MAX_OPEN_FILES, DEFAULT_NUM_RETRIES, handlerFactories);
     }
     
-    public BufferedFileBackedSortedSet(Comparator<? super E> comparator, int bufferPersistThreshold, int maxOpenFiles,
-                    SortedSetFileHandlerFactory handlerFactory) {
+    public BufferedFileBackedSortedSet(Comparator<? super E> comparator, int bufferPersistThreshold, int maxOpenFiles, int numRetries,
+                    List<SortedSetFileHandlerFactory> handlerFactories) {
         this.comparator = comparator;
-        this.handlerFactory = handlerFactory;
+        this.handlerFactories = handlerFactories;
         this.bufferPersistThreshold = bufferPersistThreshold;
+        this.numRetries = numRetries;
         this.maxOpenFiles = maxOpenFiles;
     }
     
+    private SortedSetFileHandler createFileHandler(SortedSetFileHandlerFactory handlerFactory) throws IOException {
+        boolean isCacheDirValid = false;
+        if (handlerFactory instanceof HdfsBackedSortedSet.SortedSetHdfsFileHandlerFactory) {
+            IvaratorCacheDir ivaratorCacheDir = ((HdfsBackedSortedSet.SortedSetHdfsFileHandlerFactory) handlerFactory).getIvaratorCacheDir();
+            
+            FsStatus fsStatus = null;
+            try {
+                fsStatus = ivaratorCacheDir.getFs().getStatus();
+            } catch (IOException e) {
+                log.warn("Unable to determine status of the filesystem: " + ivaratorCacheDir.getFs());
+            }
+            
+            // determine whether this fs is a good candidate
+            if (fsStatus != null) {
+                long availableStorageMB = fsStatus.getRemaining() / 0x100000L;
```

Comment:
@ivakegg 
![told you](https://i.chzbgr.com/full/6177711872/hFAF3BAD0/im-not-saying-i-told-you-so-but-i-told-you-so)

Reply:
You got me....awesome

Reply:
Since this is MB and not MiB, I think you should be dividing by 100000

Reply:
Oh, who are you, NIST???  I'll fix it, haha.

Reply:
fixed.